exporters/otlp/otlpgrpc/otlp_integration_test.go

```diff
@@ -302,6 +305,89 @@ func TestNewExporter_withHeaders(t *testing.T) {
 	assert.Equal(t, "value1", headers.Get("header1")[0])
 }
 
+func TestNewExporter_WithTimeout(t *testing.T) {
+	tts := []struct {
+		name    string
+		fn      func(exp *otlp.Exporter) error
+		timeout time.Duration
+		metrics int
+		spans   int
+		code    codes.Code
+		delay   bool
+	}{
+		{
+			name: "Timeout Spans",
+			fn: func(exp *otlp.Exporter) error {
+				return exp.ExportSpans(context.Background(), []*sdktrace.SpanSnapshot{{Name: "timed out"}})
+			},
+			timeout: time.Millisecond * 100,
+			code:    codes.DeadlineExceeded,
+			delay:   true,
+		},
+		{
+			name: "Timeout Metrics",
+			fn: func(exp *otlp.Exporter) error {
+				return exp.Export(context.Background(), otlptest.OneRecordCheckpointSet{})
+			},
+			timeout: time.Millisecond * 100,
+			code:    codes.DeadlineExceeded,
+			delay:   true,
+		},
+
+		{
+			name: "No Timeout Spans",
+			fn: func(exp *otlp.Exporter) error {
+				return exp.ExportSpans(context.Background(), []*sdktrace.SpanSnapshot{{Name: "timed out"}})
+			},
+			timeout: time.Minute,
+			spans:   1,
+			code:    codes.OK,
+		},
+		{
+			name: "No Timeout Metrics",
+			fn: func(exp *otlp.Exporter) error {
+				return exp.Export(context.Background(), otlptest.OneRecordCheckpointSet{})
+			},
+			timeout: time.Minute,
+			metrics: 1,
+			code:    codes.OK,
+		},
+	}
+
+	for _, tt := range tts {
+		t.Run(tt.name, func(t *testing.T) {
+
+			mc := runMockCollector(t)
+			if tt.delay {
+				mc.traceSvc.delay = time.Second * 10
+				mc.metricSvc.delay = time.Second * 10
+			}
+			defer func() {
+				_ = mc.stop()
+			}()
+
+			ctx := context.Background()
+			exp := newGRPCExporter(t, ctx, mc.endpoint, otlpgrpc.WithTimeout(tt.timeout))
+			defer func() {
+				_ = exp.Shutdown(ctx)
+			}()
+
+			err := tt.fn(exp)
+
+			if tt.code == codes.OK {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				s := status.Convert(err)
+				require.Equal(t, codes.DeadlineExceeded, s.Code())
+			}
```

Comment:
The code can be verified in both situations, right?

```suggestion
			}
			assert.Equal(t, tt.code, status.Convert(err).Code())
```

Reply:
Yes, thanks!

til: `status.Status` returns `codes.OK` when calling `Code()` from a nil pointer.